golang/x/relay/types/requests.go

```diff
@@ -1,9 +1,47 @@
 package types
 
+import (
+	"encoding/hex"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// RequestID is an 8 byte id used to store requests
+type RequestID [8]byte
+
+// ProofRequest is info about a proof request
 type ProofRequest struct {
 	Spends      Hash256Digest `json:"spends"`
 	Pays        Hash256Digest `json:"pays"`
 	PaysValue   uint64        `json:"paysValue"`
 	ActiveState bool          `json:"activeState"`
 	NumConfs    uint8         `json:"numConfs"`
 }
+
+// NewRequestID instantiates a RequestID from a byte slice
+func NewRequestID(b []byte) (RequestID, sdk.Error) {
+	var h RequestID
+	copied := copy(h[:], b)
+	if copied != 8 || len(b) != 8 {
```

Comment:
think the second check won't run. if `len(b) > 8` then copied will be 8.

probably `if copied != 8 || len(b) > 8`

Reply:
When I was testing it, I tried it with a value that was 9 bytes long and this check failed.  That's why I added the second check.  I will look into it.

Reply:
could also implement it before the copy as `if len(b) != 8`